### ISSUE TITLE
[no-shadow] Shows an error on enum's member if there is a variable or a type with the same name

### ISSUE THREAD

**Issue by timocov** [typescript-eslint/typescript-eslint#4310]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/main/docs/linting/TROUBLESHOOTING.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-shadow": ["error"]
  }
}
```

```TS
const time = Date.now();

enum Enum {
    time,
}
```

**Expected Result**

No error since you cannot use `Enum.time` without specifying `Enum` in the code.

**Actual Result**

```
4:2  error  'time' is already declared in the upper scope  @typescript-eslint/no-shadow
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.7.0` |
| `@typescript-eslint/parser`        | `5.7.0` |
| `TypeScript`                       | `4.5.4` |
| `ESLint`                           | `8.4.1` |
| `node`                             | `16.13.1` |

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/issues/2721#issuecomment-718804636

**Comment by timocov** [comment]
Ah dear, makes sense! Sorry for bothering and not properly searching similar issues, my bad.

**Comment by timocov** [comment]
Btw, what if it shadows a type? It seems that it is not possible to mix types and values in the same place? Like this

**index.ts**:
```ts
import { Time } from './time';

enum Enum {
    Time,
}
```

**time.ts**:
```ts
export interface Time {}
```

or to handle this the rule should use enabled type-check?

**Comment by bradzacher** [comment]
> Btw, what if it shadows a type? It seems that it is not possible to mix types and values in the same place?

Enum members do not shadow types, only values.

https://typescript-eslint.io/play/#ts=4.5.2&sourceType=module&code=PQKgBApgzgNglgOwC5gAJIJ4AdoGMBOcWSAtNPMsAgPYlQAWAhgCbUDuAXGAOQT77V83MCGAAoMZhxgAKtghgAvGACMAbgm5qCKCgBqjGAFcFygEwaxEBEYC2YAKI37AbzFgPs+QBp3ng8YQvgC+QA&rules=N4XyA&tsConfig=N4XyA

**Comment by armano2** [comment]
While importing types you should consider adding `type` keyword, with that rule correctly treats imported value as a type

```ts
import type {Type} from ''
```

or 

```ts
import {type Type} from ''
```
